system-int-test/test/cmr/system_int_test/ingest/variable_ingest_test.clj

```diff
@@ -9,23 +9,82 @@
    [cmr.system-int-test.system :as s]
    [cmr.system-int-test.utils.ingest-util :as ingest]
    [cmr.system-int-test.utils.metadata-db-util :as mdb]
-   [cmr.system-int-test.utils.variable-util :as variable-util]))
+   [cmr.system-int-test.utils.variable-util :as variable-util]
+   [cmr.umm-spec.models.umm-variable-models :as umm-v]))
 
-(use-fixtures :each (ingest/reset-fixture {"provguid1" "PROV1"}
+(use-fixtures :each (ingest/reset-fixture {"provguid1" "PROV1"
+                                           "provguid2" "PROV2"}
                                           {:grant-all-ingest? false}))
 
 (deftest variable-ingest-test
   (let [{token :token} (variable-util/setup-update-acl
-                        (s/context) "PROV1" "user1" "update-group")]
+                        (s/context) "PROV1" "user1" "update-group")
+        {token2 :token} (variable-util/setup-update-acl
+                         (s/context) "PROV2" "user1" "update-group")]
+
     (testing "ingest of a new variable concept"
-      (let [concept (variable-util/make-variable-concept)
+      (let [concept (variable-util/make-variable-concept
+                     {:Dimensions [(umm-v/map->DimensionType {:Name "Solution_3_Land"
+                                                              :Size 3
+                                                              :Type "OTHER"})]
+                      :AcquisitionSourceName "Instrument1"}
+                     {:native-id "var1"})
             {:keys [concept-id revision-id]} (variable-util/ingest-variable
-                                               concept
-                                               (variable-util/token-opts token))]
+                                              concept
+                                              (variable-util/token-opts token))
+            var-concept-id concept-id]
         (is (mdb/concept-exists-in-mdb? concept-id revision-id))
-        (is (= 1 revision-id))))
+        (is (= 1 revision-id))
+        (is (= "7bec71103f3eada1f539acb9f1072fd7"
+               (get-in (mdb/get-concept concept-id revision-id) [:extra-fields :fingerprint])))
+
+        (testing "ingest the same concept on a different provider is OK"
```

Comment:
Right - so this test seems to confirm that the unique-field constraint query is using the provider.

Reply:
Good point. I have added provider_id to the index.